W-19714749-mulesoft-inference-connector-1.2.0-minor-release-sis (#3008)
mulesoft/docs-connectors: mulesoft-inference/1.2/modules/ROOT/pages/mulesoft-inference-connector-examples.adoc

                                    'order_id': {
                                        'type': 'string',
                                        'description': 'The customer\'s order ID.'
                                    }
                                },
                                'required': ['order_id']
                            }
                        }
                    }
                ]
            }]"
				doc:name="Set Variable"
				doc:id="9b80bd71-922a-4739-804b-29ac2ba7ef7a"
				variableName="testPayload" />
		<ms-inference:tools-native-template
				doc:name="[Tools] Native Template (Reasoning only)"
				doc:id="e87a2e5e-b19b-4c2d-a4a0-8e991291de97"
				config-ref="OpenAI_Text_generation_config">
			<ms-inference:template><![CDATA[#[vars.testPayload.template]]]></ms-inference:template>
			<ms-inference:instructions><![CDATA[#[vars.testPayload.instructions]]]></ms-inference:instructions>
			<ms-inference:data><![CDATA[#[vars.testPayload.dataset]]]></ms-inference:data>
			<ms-inference:tools><![CDATA[#[vars.testPayload.tools]]]></ms-inference:tools>
		</ms-inference:tools-native-template>
	</flow>
</mule>
----

> == Example: MCP Tools

Reviewer: need to add one example for Additional request attributes

Reviewer: Intro: Many users required the ability to control the formatting of LLM outputs directly from the connector, but previously there was no straightforward way to include a response_format in requests. With this additional request attribute feature, the response_format can now be added as an request attribute.
For example: As shown below, Users can directly pass a response_format JSON object for any operation, and it will be included in the request sent to the Inference API, enabling consistent and structured outputs.

                                    'order_id': {
                                        'type': 'string',
                                        'description': 'The customer\'s order ID.'
                                    }
                                },
                                'required': ['order_id']
                            }
                        }
                    }
                ]
            }]"
				doc:name="Set Variable"
				doc:id="9b80bd71-922a-4739-804b-29ac2ba7ef7a"
				variableName="testPayload" />
		<ms-inference:tools-native-template
				doc:name="[Tools] Native Template (Reasoning only)"
				doc:id="e87a2e5e-b19b-4c2d-a4a0-8e991291de97"
				config-ref="OpenAI_Text_generation_config">
			<ms-inference:template><![CDATA[#[vars.testPayload.template]]]></ms-inference:template>
			<ms-inference:instructions><![CDATA[#[vars.testPayload.instructions]]]></ms-inference:instructions>
			<ms-inference:data><![CDATA[#[vars.testPayload.dataset]]]></ms-inference:data>
			<ms-inference:tools><![CDATA[#[vars.testPayload.tools]]]></ms-inference:tools>
		</ms-inference:tools-native-template>
	</flow>
</mule>
----

== Example: MCP Tools

> This Mule application showcases the use of the *[MCP] Tooling* operation to interact with OpenAI via the MCP connector. This operation enables you to execute tools hosted on an MCP server directly through user-defined instructions. In this example, it is configured to use the tools to check the inventory status for a specific part number, MULETEST0.

Reviewer 2: ```suggestion
This Mule application showcases the use of the *[MCP] Tooling* operation to interact with OpenAI via the MCP Connector. This operation enables you to execute tools hosted on an MCP server directly through user-defined instructions. In this example, the operation uses the tools to check the inventory status for a specific part number, `MULETEST0`.
```

                ---
                {
                    payload: payload,
                    attributes: attributes
                }]]>
			</ee:set-payload>
		</ee:message>
	</ee:transform>
</flow>

<!-- MCP client configs -->
<mcp:client-config name="MCP_Client_SSE_ERP" doc:name="MCP Client" doc:id="fd3bb75d-82e1-4aed-a1e3-f7c7ca6e28fb" clientName="MCP_Client_SSE_ERP" clientVersion="1.0.0">
        <mcp:sse-client-connection requestTimeout="60" serverUrl="${mcp.erpServer}" />
 </mcp:client-config>
    

<mcp:client-config name="MCP_Client_HTTP" doc:name="MCP Client" doc:id="163c2f4f-86bd-4860-b3b1-5606259f324f" clientName="MCP_Client_HTTP" clientVersion="1.0.0">
        <mcp:streamable-http-client-connection serverUrl="${mcp.httpServer}">
            <mcp:default-request-headers>
                <mcp:default-request-header key="client_id" value="${mcp.httpServer.clientId}" />
                <mcp:default-request-header key="client_secret" value="${mcp.httpServer.clientSecret}" />
            </mcp:default-request-headers>
        </mcp:streamable-http-client-connection>
    </mcp:client-config>
----

== Example: Additional Request Attributes

> This feature showcases the ability to control the formatting of LLM outputs directly from the connector using the *Additional Request Attributes* parameter. With this parameter, you can now specify a JSON object with attributes that will be flattened into the root level of the request payload. In this example, you can directly pass a `response_format` with type `json_schema` followed by the corresponding JSON schema. This will be included in the request sent to the Inference API, enabling consistent and structured output in the provided JSON schema. 

Reviewer 2: ```suggestion
This feature showcases the ability to control the formatting of LLM outputs directly from the connector using the *Additional Request Attributes* parameter. With this parameter, you can now specify a JSON object with attributes that are flattened into the root level of the request payload. In this example, you can directly pass a `response_format` with type `json_schema` followed by the corresponding JSON schema. This is included in the request sent to the Inference API, enabling consistent and structured output in the provided JSON schema. 
```
